synapse/lib/stormlib/todo.py

```diff
@@ -0,0 +1,75 @@
+import synapse.exc as s_exc
+import synapse.common as s_common
+
+import synapse.lib.storm as s_storm
+import synapse.lib.stormtypes as s_stormtypes
+
+@s_stormtypes.registry.registerType
+class Todo(s_stormtypes.Prim):
+    _storm_typename = 'todo'
+    _storm_locals = (
+        {'name': 'name', 'desc': 'The name of the todo.', 'type': 'str'},
+        {'name': 'args', 'desc': 'The arguments of the todo.', 'type': 'list'},
+        {'name': 'kwargs', 'desc': 'The keyword arguments of the todo.', 'type': 'dict'},
+    )
+
+    def __init__(self, runt, valu):
+        s_stormtypes.Prim.__init__(self, None)
+        self.runt = runt
+
+        if not isinstance(valu, (list, tuple)):
+            raise s_exc.BadArg(mesg='Todo value must be a list or tuple')
+
+        self.name = valu[0]
+        self.args = valu[1] if len(valu) > 1 and isinstance(valu[1], (list, tuple)) else (valu[1],) if len(valu) > 1 else ()
+        self.kwargs = valu[2] if len(valu) > 2 and isinstance(valu[2], dict) else {}
+
+        self.locls.update({
+            'name': self.name,
+            'args': self.args,
+            'kwargs': self.kwargs,
+        })
+
+    def value(self):
+        return (self.name, self.args, self.kwargs)
+
+@s_stormtypes.registry.registerLib
+class LibTodo(s_stormtypes.Lib):
+    '''
+    A Storm library for parsing todo strings.
+    '''
+    _storm_locals = (
+        {'name': 'parse', 'desc': 'Parse a todo string into a todo object.',
+         'type': {'type': 'function', '_funcname': 'parse',
+            'args': (
+                {'name': 'valu', 'type': 'str', 'desc': 'The todo string to parse into a todo object.'},
+            ),
+            'returns': {'type': 'todo', 'desc': 'A todo object.'},
+        }},
+    )
+    _storm_lib_path = ('todo',)
+
+    def getObjLocals(self):
+        return {
+            'parse': self.parse,
+        }
+
+    async def parse(self, valu):
+        todo = await s_stormtypes.toprim(valu)
```

Comment:
This should just mirror the functionality of common.todo(). Introducing the expectation that it parses command lines is out of scope and introduces in-band control flow issues and parser ambiguity. It should be used by API callers only.

Reply:
Yeah I wasn't sure where the guardrails were and attempted to make the `todo` compilation easier. I'll rejig.

Reply:
Maybe we can also move this into LibBase so the function becomes `$lib.todo(...)`?

Reply:
This was discussed today at stand-up. Going to move it to `$lib.utils.todo(...)` with updated test cases. Will be in the next commit.